sys/vfs.c

```diff
@@ -191,10 +191,10 @@ int vfs_lookup(const char *path, vnode_t **vp) {
 
   /* Copy path into a local buffer, so that we may process it. */
   size_t n = strlen(path);
-  if (n >= VFS_PATH_MAX)
+  if (n >= PATH_MAX)
     return -ENAMETOOLONG;
-  char pathcopy[VFS_PATH_MAX];
-  strlcpy(pathcopy, path, VFS_PATH_MAX);
+  char pathcopy[PATH_MAX];
```

Comment:
Because `PATH_MAX` is rather large compared to kernel thread stack size (4096 bytes) please move the allocation to heap by using `M_TEMP` memory pool.